proto/proto/store/rpc.proto

```diff
@@ -292,3 +295,48 @@ message SyncNotesResponse {
     // List of all notes together with the Merkle paths from `response.block_header.note_root`.
     repeated note.NoteSyncRecord notes = 4;
 }
+
+// SYNC STORAGE MAP
+// ================================================================================================
+
+// Storage map synchronization request.
+//
+// Allows clients to sync storage map values for specific public accounts within a block range,
+// with support for cursor-based pagination to handle large storage maps.
+message SyncStorageMapsRequest {
+    // Block number to start sending updates from (inclusive).
+    fixed32 block_from = 1;
+
+    // Block number up to which to sync. If not specified, syncs up to the latest block.
+    //
+    // If specified, this block must be close to the chain tip (i.e., within 30 blocks),
+    // otherwise an error will be returned.
+    optional fixed32 block_to = 2;
+    
+    // Account for which we want to sync storage maps.
+    account.AccountId account_id = 3;
+}
+
+message SyncStorageMapsResponse {
+    // The block number of the last row included in this response.
+    // For chunked responses, this may be less than request.block_to.
+    fixed32 block_num = 1;
+
+    // Current chain tip
+    fixed32 chain_tip = 2;
+
+    // The list of storage map updates.
+    repeated StorageMapUpdate updates = 3;
```

Comment:
The set of returned updates here could contain duplicates, right? If so, I would make it clear in the comment that only the latest value for a given `(slot_index, key)` combination should be retained.